scimath/Mathematics/ClassicalStatistics.tcc

```diff
@@ -2352,10 +2403,10 @@ void ClassicalStatistics<CASA_STATP>::_makeBins(
 #define _minMaxCode \
     if (! mymin.null()) { \
         if (*datum < *mymin) { \
-            mymin = new AccumType(*datum); \
+            *mymin = *datum; \
         } \
         else if (*datum > *mymax) { \
```

Comment:
one should remove the else here as that prevents using branchless instructions and vectorization.
Its the classic example of the difference in what you think is the most efficient code but is not how a x86 cpu actually works.
x86 has min and max instructions that do not need a branch, by placing a else if here instead of unconditionally assigning the maximum you force the compiler to put a branch there as in the (possible) presence of threads it cannot remove conditional assignments.

but then these are shared pointers so I don't actually know what the compiler will do with it :/